### DISCUSSION TITLE
github.com or Desktop App : select branch name for copying to Windows clipboard?

### DISCUSSION THREAD

**Discussion by Brad-Stiritz** [community/community#24022] [New to GitHub]
<p>Hi everyone,</p>
<p>The repositories I work on all use meaningful, often verbose branch names.  I frequently need to reference branch names, e.g. within repository Issues.</p>
<p>Problem: I’m not finding a way to select and right-click-copy a branch name, from the dropdown branch list,  either on <a href="http://github.com" rel="nofollow noopener">github.com</a> or in Windows Github Desktop.</p>
<p>The error-prone workaround is to manually type out the branch name.</p>
<p>Any help appreciated, please?</p>
<p>Thanks,</p>
<p>Brad</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/brad-stiritz">@brad-stiritz</a>,</p>
<p>I just experiemented with this myself, and I found that if I held down “option/alt” while clicking and dragging on a branch name in the dropdown, I was able to select the text there and then “cmnd/ctrl+c” to copy. </p>
<p>I hope this helps!</p>

**Comment by Brad-Stiritz** [comment]
<p>Hi Nadia,</p>
<p>Thank you very much for sharing this very cool solution. I never suspected that the <code>Alt</code> key could be used in this way.</p>
<p>Sincere thanks and best wishes,</p>
<p>Brad</p>

**Comment by marin-takanov** [comment]
<p>I’m still unable to find a good solution as everything is a workaround of this common “problem”. It would be nice if the GitHub team just add a button for coping the branch name.</p>
<p>Until then, I will share other ways you can copy the branch name:</p>
<ul>
<li>
<p>from <a href="https://github.com/%7Borganization%7D/%7Brepository%7D/branches" rel="noopener nofollow ugc">https://github.com/{organization}/{repository}/branches</a>, search the branch you want to copy by writing the first one or two unique words it contain until you see it. Hold ctrl and left click  with the mouse on the branch name. This will prompt a context menu, click Copy. The fast way is to ctrl + double left click so you instantly close the context menu and select the branch name.</p>
</li>
<li>
<p>from <a href="https://github.com/%7Borganization%7D/%7Brepository%7D/tree/%7Bbranch_name%7D" rel="noopener nofollow ugc">https://github.com/{organization}/{repository}/tree/{branch_name}</a>, just copy the branch name from the URL</p>
</li>
</ul>

**Reply by RamtinFD** [reply]
It's 2026 and this is still a "problem"!  😏
